nodejs/dist/uws.js

```diff
@@ -110,15 +119,13 @@ class Server extends EventEmitter {
             });
 
             this.nativeServer.onDisconnection((nativeSocket, socket) => {
-                if (socket.disconnectionCallback) {
-                    socket.disconnectionCallback();
-                }
+                socket.disconnectionCallback();
                 /* make sure to clear any set data */
                 this.nativeServer.setData(nativeSocket);
             });
 
             this.nativeServer.onMessage((nativeSocket, message, binary, socket) => {
-                if (socket.messageCallback) {
+                if (socket.messageCallback !== noop) {
```

Comment:
Technically, we don't need this if-statement, since the callbacks are always callable, but just in case to avoid calling `message.toString()` if there is no real listener, I kept this if-statement. We could of course say "we always expect there to be relevance in listening for messages, so the overhead of the if-statement is more painful than the overhead of `message.toString()` being called for nothing.